src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -308,7 +308,7 @@ TxQ::canBeHeld(STTx const& tx, OpenView const& view,
             an early one fail or get dropped.
         */
         canBeHeld = accountIter == byAccount_.end() ||
-            !replacementIter ||
+            replacementIter ||
```

Comment:
Yes, the old code was a bug. Your analysis is correct. Here's the new test which demonstrates the issue, and which fails if this change is removed. https://github.com/ximinez/rippled/blob/b98c6154c58d48431c7da6558cec003ba9b44e95/src/ripple/app/tests/TxQ_test.cpp#L835-L845 I also realize that that test needs to be expanded to actually do a replacement when the user has 10 txs in the queue.